pages/library/_viewId.vue

```diff
@@ -157,6 +198,127 @@ export default Vue.extend({
+            includeItemTypes: '',
+            recursive: true,
+            sortBy: this.orderMethod,
+            sortOrder: this.sortDirection,
+            fields: this.orderMethod
+          };
+
+          if (collectionInfo.data.Items[0].CollectionType === 'tvshows') {
+            options.includeItemTypes = 'Series';
+          } else if (collectionInfo.data.Items[0].CollectionType === 'movies') {
+            options.includeItemTypes = 'Movie';
+          } else if (collectionInfo.data.Items[0].CollectionType === 'books') {
+            options.includeItemTypes = 'Book';
+          }
+
+          const filterString = this.makeFilterString(
+            this.allFilters.filters.items,
+            ','
+          );
+          options.filters = filterString;
+
+          for (let i = 0; i < this.allFilters.features.items.length; i++) {
+            const feature = this.allFilters.features.items[i];
+            if (!feature.selected) {
+              continue;
+            }
+
+            options[feature.value] = true;
+          }
+
+          const genreString = this.makeFilterString(
+            this.allFilters.genres.items,
+            '|'
+          );
+          options.genres = genreString;
+
+          const ratingString = this.makeFilterString(
+            this.allFilters.officialRatings.items,
+            '|'
+          );
+          options.officialRatings = ratingString;
+
+          let videoTypeString = '';
+          for (let i = 0; i < this.allFilters.videoTypes.items.length; i++) {
+            const videoType = this.allFilters.videoTypes.items[i];
+            if (!videoType.selected) {
+              continue;
+            }
+            if (videoType.label === 'SD') {
+              options.isHd = false;
+              continue;
+            }
+            if (videoType.label === 'HD') {
+              options.isHd = true;
+              continue;
+            }
+            if (videoType.label === '4K' || videoType.label === '3D') {
+              options[videoType.value] = true;
+            } else {
+              if (videoTypeString.length > 0) {
+                videoTypeString += ',';
+              }
+              videoTypeString += videoType.value;
+            }
+          }
+          options.videoTypes = videoTypeString;
+
+          const yearString = this.makeFilterString(
+            this.allFilters.years.items,
+            ','
+          );
+          options.years = yearString;
+
+          const itemsResponse = await this.$api.items.getItems(options);
+
+          if (itemsResponse.data) {
+            this.loaded = true;
+          }
+
+          this.items = itemsResponse.data.Items || [];
+        }
+      } catch (error) {
+        // Can't get given library ID
+        this.$nuxt.error({
+          statusCode: 404,
+          message: this.$t('libraryNotFound') as string
+        });
+      }
+    },
+    makeFilterString(filterList: FilterItem[], seperator: string) {
```

Comment:
Similarly, this should live in the filter menu component, since it's very much tied to that.